lib/src/mock_query.dart

```diff
@@ -90,7 +90,7 @@ class MockQuery extends Mock implements Query {
       List<dynamic> whereIn,
       bool isNull}) {
     _QueryOperation operation = (documents) => documents
-        .where((document) => _valueMatchesQuery(document[field],
+        .where((document) => _valueMatchesQuery(_getValueFromDocument(document, field),
```

Comment:
Your `_getValueFromDocument` utility method is very similar to [util.dart's getSubPath](https://github.com/atn832/cloud_firestore_mocks/blob/45b6837fd614389adf30bf02e4c3b614a1ff1c49/lib/src/util.dart#L3), except that in your case, the separator is `'.'` instead of `'/'`, and yours will throw an exception if the path does not exist.

Instead of making your own method, could you tweak util.dart's getSubPath to accept an optional `separator` argument and reuse it? It'll remove the code duplication and handle missing paths for you.

Reply:
Also in order to use `getSubPath`, you'll probably need to pass `document.data` instead of document and cast the field to `String`: `getSubPath(document.data, field as String)`.

Btw is `Query.where` ever used with a non-string field? What's the expected behavior then?